### PR TITLE
[Close loop BZ: 2239208] Test case to create PVC in megabytes

### PR DESCRIPTION
**Pull request by nagendra202**
Link to BZ : [2239208](https://bugzilla.redhat.com/show_bug.cgi?id=2239208)

This  new test case will take care of:
1. Creating Cephfs PVC with small size 499M, 500M, 501M.
2. Passing size_unit as 'M' [mega-bytes] and the created PVC size will be displayed in 'Mi' format in pvc yaml.
3. Verifying the created PVC size by converting it from 'Mi' [mebi-bytes] to 'bytes' and comparing it with the given size.
4. The expectation here is the created PVC sized should be greater than or equal to the given size. It should not be lesser than the given size.

### CODE REVIEW

File: tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
+import pytest
+import logging
+from ocs_ci.framework.testlib import bugzilla, tier1
+from ocs_ci.ocs import constants
+from ocs_ci.framework.pytest_customization.marks import magenta_squad
+
+log = logging.getLogger(__name__)
+
+
+@pytest.mark.parametrize(
```
Comment (OdedViner) [inline tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py:10]: Why do you use `parametrize` if you have only one test case?

File: tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
+import pytest
+import logging
+from ocs_ci.framework.testlib import bugzilla, tier1
```
Comment (PrasadDesala) [inline tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py:3]: add an empty line after this

File: tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    argnames="interface",
+    argvalues=[
+        pytest.param(
+            *[constants.CEPHFILESYSTEM], marks=pytest.mark.polarion_id("OCS-323")
+        ),
+    ],
+)
+@tier1
+@bugzilla("2239208")
+@magenta_squad
```
Comment (PrasadDesala) [inline tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py:20]: this should be in greensquad according to the test placement

File: tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+    def test_pvc_creation_in_megabytes(self, interface, pvc_factory):
+        """
+        This function facilitates
+        1. Create PVC in Cephfs with size_unit as "M" . Used access mode is RWX.
+        2. Convert the PVC's size from 'Mi' to bytes.
+        3. Validate created PVC size with the given_pvc_size in "M" format after converting it into bytes.
+        """
+
+        access_mode = constants.ACCESS_MODE_RWX
```
Comment (PrasadDesala) [inline tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py:39]: RWX is a must for this issue?

File: tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+from ocs_ci.framework.pytest_customization.marks import magenta_squad
+
+log = logging.getLogger(__name__)
+
+
+@pytest.mark.parametrize(
+    argnames="interface",
+    argvalues=[
+        pytest.param(
+            *[constants.CEPHFILESYSTEM], marks=pytest.mark.polarion_id("OCS-323")
```
Comment (PrasadDesala) [inline tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py:14]: please add the correct polarion id

File: tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+            logging.info(f"PVC created with {actual_pvc_size} capacity")
+            # conversion formulae for 'M' to bytes
+            mega_bytes_to_bytes = 1000 * 1000
+            # conversion formulae for 'Mi' to bytes
+            mebi_bytes_to_bytes = 1024 * 1024
+
+            if (int(actual_pvc_size[:-2]) * mebi_bytes_to_bytes) >= (
+                int(size) * mega_bytes_to_bytes
+            ):
+                log.info("PVC created correctly")
```
Comment (PrasadDesala) [inline tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py:59]: please elaborate the log message explaining the success message using fstring

File: tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+            else:
+                log.error("Created PVC size didn't met the given size")
```
Comment (PrasadDesala) [inline tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py:61]: you can delete this line as you have an assert explaining the failure

File: tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+            log.info(f"Creating {interface} based PVC with the given size {size}M")
+            # Creating PVC using pvc_factory
+            pvc_obj = pvc_factory(
+                interface=interface, access_mode=access_mode, size=size, size_unit="M"
+            )
+            # getting pvc yaml using pvc_obj.get() and getting the pvc capacity from the yaml
+            actual_pvc_size = pvc_obj.get().get("status").get("capacity").get("storage")
+            logging.info(f"PVC created with {actual_pvc_size} capacity")
+            # conversion formulae for 'M' to bytes
+            mega_bytes_to_bytes = 1000 * 1000
```
Comment (hnallurv) [inline tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py:52]: there is no need to keep this and next calculation inside the loop. Please move them out

File: tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py (hunk 1 above)
Comment (nagendra202) [reply tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py]: made changes

File: tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py (hunk 2 above)
Comment (nagendra202) [reply tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py:3]: done

File: tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py (hunk 3 above)
Comment (nagendra202) [reply tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py]: changed squad to green_squad

File: tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py (hunk 5 above)
Comment (nagendra202) [reply tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py]: done. -->@pytest.mark.polarion_id("OCS-5476")

File: tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py (hunk 6 above)
Comment (nagendra202) [reply tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py]: done

Example:
INFO  - Created PVC capacity 503316480 bytes where the requested capacity was 501000000 bytes

File: tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py (hunk 7 above)
Comment (nagendra202) [reply tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py]: done

File: tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py (hunk 8 above)
Comment (nagendra202) [reply tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py]: done, moved those two lines of code to outside of for loop.

File: tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py (hunk 2 above)
Comment (PrasadDesala) [reply tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py:3]: sorry, I meant empty line after line2

File: tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+
+    def test_pvc_creation_in_megabytes(self, pvc_factory):
+        """
+        This function facilitates
+        1. Create PVC in Cephfs with size_unit as "M" . Used access mode is RWX.
+        2. Convert the PVC's size from 'Mi' to bytes.
+        3. Validate created PVC size with the given_pvc_size in "M" format after converting it into bytes.
+        """
+        access_mode = constants.ACCESS_MODE_RWX
+        interface = constants.CEPHFILESYSTEM
```
Comment (PrasadDesala) [inline tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py:33]: same as my above comment - you dont need this var, you can directly pass it to the func

File: tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+        given_pvc_size = ["499", "500", "501"]
```
Comment (PrasadDesala) [inline tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py:34]:
```suggestion
        given_pvc_sizes_mb = ["499", "500", "501"]
```

File: tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+        # conversion formulae for 'M' to bytes
+        mega_bytes_to_bytes = 1000 * 1000
+        # conversion formulae for 'Mi' to bytes
+        mebi_bytes_to_bytes = 1024 * 1024
+        for size in given_pvc_size:
```
Comment (PrasadDesala) [inline tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py:39]:
```suggestion
        for size_mb in given_pvc_size_mb:
```

File: tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+            log.info(f"Creating {interface} based PVC with the given size {size}M")
+            # Creating PVC using pvc_factory
+            pvc_obj = pvc_factory(
+                interface=interface, access_mode=access_mode, size=size, size_unit="M"
+            )
+            # getting pvc yaml using pvc_obj.get() and getting the pvc capacity from the yaml
+            actual_pvc_size = pvc_obj.get().get("status").get("capacity").get("storage")
+            log.info(f"PVC created with the capacity of {actual_pvc_size} ")
```
Comment (PrasadDesala) [inline tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py:47]:
```suggestion
            log.info(f"PVC created with the capacity of size {actual_pvc_size} ")
```

File: tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+            created_pvc_in_bytes = int(actual_pvc_size[:-2]) * mebi_bytes_to_bytes
```
Comment (PrasadDesala) [inline tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py:48]:
```suggestion
            created_pvc_size_bytes = int(actual_pvc_size[:-2]) * mebi_bytes_to_bytes
```

File: tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py [hunk 14]
```diff
@@ -0,0 +1,57 @@
...
+            requested_pvc_in_bytes = int(size) * mega_bytes_to_bytes
```
Comment (PrasadDesala) [inline tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py:49]:
```suggestion
            requested_pvc_size_bytes = int(size_mb) * mega_bytes_to_bytes
```

File: tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py [hunk 15]
```diff
@@ -0,0 +1,57 @@
...
+            if created_pvc_in_bytes >= requested_pvc_in_bytes:
+                log.info(
+                    f"Created PVC {created_pvc_in_bytes} bytes where the requested was {requested_pvc_in_bytes} bytes"
+                )
+            else:
+                assert (
+                    False
+                ), f"Actual PVC size {actual_pvc_size} is different than the given PVC size {size}M"
```
Comment (PrasadDesala) [inline tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py:57]:
```suggestion
            assert created_pvc_size_bytes >= requested_pvc_size_bytes, 
                f"Actual PVC size {created_pvc_size_bytes} bytes is less than the given PVC size {requested_pvc_size_bytes}"
            
            logger.info(
                f"Created PVC {created_pvc_size_bytes} bytes where the requested was {requested_pvc_size_bytes} bytes"
            )
```

**Review by hnallurv — DISMISSED** [review]

**Review by OdedViner — APPROVED** [review]

**Review by hnallurv — APPROVED** [review]

File: tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py [hunk 16]
```diff
@@ -0,0 +1,59 @@
...
+from ocs_ci.framework.pytest_customization.marks import green_squad
+
+log = logging.getLogger(__name__)
+
+
+@tier1
+@bugzilla("2239208")
+@green_squad
+@pytest.mark.polarion_id("OCS-5476")
+class TestPvcCreationInMegabytes(E2ETest):
```
Comment (vavuthu) [inline tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py:15]: you need to import E2Etest

Comment (nagendra202) [reply tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py:16]: Please check now.

File: tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py [hunk 17]
```diff
@@ -0,0 +1,60 @@
+import pytest
+import logging
+
+from ocs_ci.framework.testlib import E2ETest
+from ocs_ci.framework.testlib import bugzilla, tier1
```
Comment (vavuthu) [inline tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py:5]: combine those two into single

from ocs_ci.framework.testlib import bugzilla, E2ETest, tier1

Comment (nagendra202) [reply tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py]: done

File: tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py (hunk 16 above)
Comment (vavuthu) [reply tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py:15]: could you please order the imports

Comment (nagendra202) [reply tests/functional/pv/pv_services/test_pvc_creation_in_megabytes.py:15]: Done.

**Review by OdedViner — APPROVED** [review]

**Pull request merged**